js/ui/appDisplay.js

```diff
@@ -2382,7 +2401,10 @@ const AppIcon = new Lang.Class({
         return [sourceActor.actor];
     },
 
-    _onNewGtkNotificationSource: function(daemon, source) {
+    _sourceAdded: function(tray, source) {
+        if (!(source instanceof NotificationDaemon.GtkNotificationDaemonAppSource))
+            return;
```

Comment:
Well, we only connect to the signal for the ChatBox case. So I want to filter and bail out as early as possible. If we open it up for more apps we can remove that special casing.